dracut/driver_updates.py

```diff
@@ -625,10 +625,17 @@ def main(args):
 
     if mode in ('--disk', '--net'):
         request, dev = args
+
+        # Guess whether this is an ISO or RPM based on the filename.
+        # If neither matches, assume it is a device node and processes as an ISO.
+        # This is relevant for both --disk and --net since --disk could be
+        # pointing to files within the initramfs.
         if dev.endswith(".iso"):
             process_driver_disk(dev)
-        else:
+        elif dev.endswith(".rpm"):
```

Comment:
The part that collects the rpms is in fetch-driver-net.sh (https://github.com/rhinstaller/anaconda/commit/14c4fa914324fd60e38f608b52d1511dfd4713d1#diff-7303c2903a26e1e3529239060b08765dR50), where it loops over *.rpm and runs driver-updates for each. I'm honestly not real sure why the conditional here was reversed, unless the goal was to process things not ending in .rpm as rpm. If that's the case, a lot more is going to have to change, since right now the most common use case is busted.